lib/mix/lib/mix.ex

```diff
@@ -965,6 +983,31 @@ defmodule Mix do
     Path.join(app_dir, relative_path)
   end
 
+  defp remove_leftover_deps!(install_dir) do
+    build_lib_dir = Path.join([install_dir, "_build", "dev", "lib"])
+    deps_dir = Path.join(install_dir, "deps")
+
+    deps = build_lib_dir |> File.ls!() |> MapSet.new()
```

Comment:
Building `MapSet` for a single operation (difference) is less efficient than using `foo -- bar`. So we can rely on lists here. :) MapSet are worth if doing several.

Reply:
Another option is to build the MapSet here and use it in the comprehension to see if you should keep it or not. Then you only build one. It should be the fastest approach.

Reply:
Oh yes!